doc: thingy53: nrf52: Update docs after config simplification (USB CDC ACM, FOTA)

mkapala-nordic opened the pull request:
Updated user guides for nRF52/53 series DKs & Thingy:53 and changelog.

PR has been separated for several commits:
* link fix
* related to USB CDC ACM and pm-ext-flash configuration: #10115 & #10024
* Thingy:53 + nRF52/53 series DKs related to FOTA configuration: #9999 PR

Jira: NCSDK-19197

NordicBuilder commented:
## Test specification

### [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10256/2/)
- Skipped

### [CI/Jenkins/integration](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10256/2/)
- Skipped



*Note: This message is automatically posted and updated by the CI*

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-10256
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

kapi-no commented on doc/nrf/releases/release-notes-changelog.rst:
```diff
@@ -52,6 +52,22 @@ Build system
+
+* nRF52 and nRF53 series development kits:
+
+  * Added :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` and :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` to configure FOTA updates.
```
```suggestion
  * Added :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` and :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` to configure FOTA updates over Bluetooth Low Energy in the default setup.
    The default setup uses MCUmgr libraries with the Bluetooth transport layer and requires the user to enable MCUboot bootloader.
```

kapi-no commented on doc/nrf/releases/release-notes-changelog.rst:
```diff
@@ -52,6 +52,22 @@ Build system
+  * Simplified board configuration:
+
+    * Added :kconfig:option:`CONFIG_BOARD_SERIAL_BACKEND_CDC_ACM` and DTS `cdc_acm_uart` node to configure USB CDC ACM to be used as logger's backend by default.
+    * Provided support for Thingy:53 FOTA updates within the :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` option.
```
If the MCUboot bootloader was enabled in this release, it would be great to mention it as a separate entry. @MarekPieta

mkapala-nordic replied:
Judging by the commit date 28.11.2021, MCUboot bootloader was enabled for Thingy:53 already for NCS 2.2, but Marek please confirm just in case.

kapi-no commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -38,13 +38,14 @@ To enable support for FOTA updates, do the following:
+   * By default, :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` option is implied when FOTA is enabled using mentioned option.
+     This option automatically extends the Bluetooth buffers, which allows to speed up the FOTA transfer over Bluetooth, but can also increase RAM usage.
+     You can disable this option manually if necessary.
 
```
It would be great to add a section on how to build for a specific project like the Peripheral LBS sample. Something along these lines: 

_You can enable the FOTA feature in a single command (...) Please see how it can be done for the Peripheral LBS sample (replace the `<target>` with your target board):_

```
west build -b <target> -- -DCONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU=y
```

mkapala-nordic commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -220,23 +220,14 @@ The overlay is applied automatically.
 ==================
 
-Each sample must enable the MCUboot bootloader and use the same configuration of the bootloader as all of the Nordic Thingy:53-compatible samples in the |NCS|.
+MCUboot bootloader is enabled by default for Thingy:53 in board Kconfig file.
```
> I wonder if we should inform that these feature is only available in the NCS (this would not be enabled out-of-the-box if we use vanilla Zephyr).

To be honest, I am not really convinced about that. These docs relates to only NCS and for Vanilla Zephyr there are other docs. For now i did not addressed this comment in code.

mkapala-nordic commented:
I think I addressed most of the comments.

MarekPieta commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -34,17 +34,38 @@ You can also use FOTA updates to replace the application.
+
+   * All of the SMP command handlers mentioned in the previous paragraph.
+   * SMP BT reassembly feature.
+   * :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` option -- automatically extends the Bluetooth buffers, which allows to speed up the FOTA transfer over Bluetooth, but can also increase RAM usage.
```
`also increases` (it always increases the RAM usage actually :) )

MarekPieta commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -34,17 +34,38 @@ You can also use FOTA updates to replace the application.
+   * SMP BT reassembly feature.
+   * :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` option -- automatically extends the Bluetooth buffers, which allows to speed up the FOTA transfer over Bluetooth, but can also increase RAM usage.
+
+If necessary, any of the implied options or defaulted values by :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` can be modified.
```
`... implied options or defaulted values introduced by :kconfig:option: ...`

MarekPieta commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -34,17 +34,38 @@ You can also use FOTA updates to replace the application.
+
+If necessary, any of the implied options or defaulted values by :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` can be modified.
+
+You can either add the above Kconfig options to your :file:`prj.conf` file or inline them in a project build command.
```
to your application configuration files (there can be multiple application configuration files in Zephyr, configuration files for other build types can have different names too)

MarekPieta commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -299,37 +290,17 @@ FOTA updates on the Nordic Thingy:53 are supported using MCUmanager's Simple Man
-
-See the :file:`thingy53.c` source file in a compatible sample for how this is done.
+To enable FOTA feature, use :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option.
+For details about implementation, please take a look at :ref:`ug_nrf52_developing_ble_fota` guide.
```
I would also note that Thingy:53 supports network core upgrade by default (not supported for other nRF53 boards, because only Thingy:53 enables it in its MCUboot bootloader's default configuration).

MarekPieta commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -34,17 +34,38 @@ You can also use FOTA updates to replace the application.
+
+   * All of the SMP command handlers mentioned in the previous paragraph.
+   * SMP BT reassembly feature.
+   * :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` option -- automatically extends the Bluetooth buffers, which allows to speed up the FOTA transfer over Bluetooth, but can also increase RAM usage.
```
I would also inform that Bluetooth buffers configuration is also automatically applied to the network core child image (maybe you could mention which network core images are supported aswell - hci_rpmsg and multiprotocol_rpmsg).

mkapala-nordic replied:
I think that we should with that for technical writer's input like @kapi-no suggested in https://github.com/nrfconnect/sdk-nrf/pull/10256#discussion_r1105847614

kapi-no commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -295,37 +290,17 @@ FOTA updates on the Nordic Thingy:53 are supported using MCUmanager's Simple Man
 The application acts as a GATT server and allows the connected Bluetooth Central to perform the firmware update for both the application and network core.
 The Bluetooth configuration is updated to allow quick DFU data transfer.
 
-The application supports SMP handlers related to:
-
-* Image management.
-* Operating System (OS) management -- used to reboot the device after the firmware upload is complete.
-* Erasing settings partition -- used to ensure that a new application is not booted with incompatible content in the settings partition written by the previous application.
```
I am not sure why this section is removed

mkapala-nordic replied:
Removed that as I moved that to section in nRF52/53 to not duplicate docs.

kapi-no commented on doc/nrf/releases/release-notes-changelog.rst:
```diff
@@ -52,6 +52,27 @@ Build system
+
+  * Added :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` and :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` to configure FOTA updates over Bluetooth Low Energy in the default setup.
+    The default setup uses MCUmgr libraries with the Bluetooth transport layer and requires the user to enable MCUboot bootloader.
+    See details in :ref:`ug_nrf52_developing_ble_fota` section of development guide.
```
If there is distinct UG FOTA ref for nRF53 DK, consider updating this part

```suggestion
    See details in :ref:`ug_nrf52_developing_ble_fota` or (link to nrf53 UG section with fota) section of development guide.
```

mkapala-nordic replied:
In nrf5340.rst there is such snippet:
```
.. include:: ../nrf52/developing.rst
   :start-after: fota_upgrades_start
   :end-before: fota_upgrades_end
```
so in a nutshell it should be the same.

mkapala-nordic replied:
But as i can see in CI there is something wrong with that :ref:`ug_nrf52_developing_ble_fota` so technical writers needs to take a look at this anyway

peknis replied:
So, I edited Kamil's suggestion and it should work.

mkapala-nordic replied:
I've missed that one, pushed change right now: https://github.com/nrfconnect/sdk-nrf/compare/8924e7e3676114a5192eed70f6ee812431b5a61d..0344a073b6af2b7bfaedcf3286d594a29e5d99c5

MarekPieta commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -243,25 +243,29 @@ See :file:`thingy53_nrf5340_cpuapp.conf` in a compatible sample for how this is
 USB
 ===
 
-An application compatible with the Nordic Thingy:53 is expected to enable USB CDC ACM as backend for logging.
```
Consider removing information about applications/samples that are compatible with Thingy:53 in the NCS. Currently all of the applications and samples are built together with MCUboot so that basic compatibility with the preprogrammed bootloader and default logging method is added out-of-the box (and OTA DFU can be enabled for any sample using a single Kconfig option).

MarekPieta replied:
Also multi-image build system will be always used (application is in general built together with MCUboot bootloader and netcore image.

MarekPieta replied:
I am also not sure if we should explicitly mention building separate image for the network core (I think such image would be misaligned with Thingy:53 support in the NCS in current version). We could simply inform that we in general build for app core and net core image is automatically added as a child image. I think

See:
`The build target should be thingy53_nrf5340_cpuapp or thingy53_nrf5340_cpuapp_ns when building samples for the application core and thingy53_nrf5340_cpunet when building for the network core.` in the current documentation

MarekPieta replied:
You could also update the `Firmware update using external debug probe` section. Currently all of the samples/apps are compatible with preprogrammed MCUboot bootloader.

MarekPieta replied:
At the beginning of `Thingy:53 application guide` you could mention that in general samples and applications are compatible with preprogrammed bootloader out-of-the box and only SMP OTA DFU needs to be explicitly enabled using an additional Kconfig option (most of the configuration is done in the background and user does not need to actually do anything there).

MarekPieta commented on doc/nrf/releases/release-notes-changelog.rst:
```diff
@@ -52,6 +52,17 @@ Build system
 
   * Manifest file entry ``mbedtls`` (:makevar:`ZEPHYR_MBEDTLS_MODULE_DIR`) checked out at path :file:`modules/crypto/mbedtls` now points to |NCS|'s fork of MbedTLS instead of Zephyr's fork.
 
+Board support
```
The `pm-ext-flash` node was introduced before the previous release. I think we should not mention it in the commit message.

mkapala-nordic replied:
@peknis, I did not found better category than that.
@kapi-no's and my changes were related to board configurations of nRF52 & nRF53 DKs and Thingy:53 so i thought it would be good idea to put it under `Board support`.
As i changed also app/dev guides i added entry at the bottom of a changelog with files that i edited.

Should i change that to something like `Working with nRF52 and nRF53 Series` then?

peknis replied:
Yes, use Working with nRF52 Series, Working with nRF53 and Application development as the headings for the changes. Look at the documentation structure on the left in  https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/index.html (the first level headings). Those are the choices for headings in the Changelog. I hope @divipillai and @annwoj agree with this. Talk with these ladies for immediate help.

divipillai replied:
Yeah, agree with @peknis .

MarekPieta commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -299,37 +290,17 @@ FOTA updates on the Nordic Thingy:53 are supported using MCUmanager's Simple Man
-See the :file:`thingy53.c` source file in a compatible sample for how this is done.
+To enable FOTA feature, use :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option.
+For details about implementation, please take a look at :ref:`ug_nrf52_developing_ble_fota` guide.
+Note that MCUboot is already enabled by default on Thingy:53 and step with its configuration from guide can be ignored.
 
```
` and step with its configuration from guide can be ignored` -> `so that there is no need to do it manually in application or sample configuration`

kapi-no commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -34,17 +34,38 @@ You can also use FOTA updates to replace the application.
 FOTA over Bluetooth Low Energy
 ==============================
 
+FOTA updates are supported using MCUmanager's Simple Management Protocol (SMP) over Bluetooth.
+The application acts as a GATT server and allows the connected Bluetooth Central to perform the firmware update.
+Sample must support Bluetooth peripheral role (:kconfig:option:`CONFIG_BT_PERIPHERAL`) in order to use this feature.
+
+The application supports SMP handlers related to:
+
+* Image management.
+* Operating System (OS) management -- used to reboot the device after the firmware upload is complete.
+* Erasing settings partition -- used to ensure that a new application is not booted with incompatible content in the settings partition written by the previous application.
```
If possible, I guess this section can be shared between the Thingy:53 ug and nRF52/53 DK UGs

Let's wait with this for the technical writers input

mkapala-nordic replied:
I think so too. The only problem is the `CONFIG_BOOTLOADER_MCUBOOT` which is needed to be manually in nRF52/53 DKs and is already enabled by default fot Thingy:53

peknis replied:
So, if there is some part that is good for all of the listed docs (whether it is a whole section or just a subset of it), you can mark such area with the `.. fota_something_start` in the beginning and `.. fota_something_end` at the end of the area. Then just use that marked area in other docs the way you had in your earlier comment above.

kapi-no commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -34,17 +34,38 @@ You can also use FOTA updates to replace the application.
 * Use MCUboot as the upgradable bootloader (:kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` must be enabled).
   For more information, go to the :doc:`mcuboot:index-ncs` page.
-* Enable the mcumgr module that handles the transport protocol over Bluetooth® Low Energy as follows:
+* Enable :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option, which implies configuration of:
```
Also here

(Let's wait with this improvement suggestion for the technical writer's input)

peknis replied:
```suggestion
* Enable the :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option, which implies configuration of the following:
```

peknis commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -34,17 +34,38 @@ You can also use FOTA updates to replace the application.
+
+   * All of the SMP command handlers mentioned in the previous paragraph.
+   * SMP BT reassembly feature.
+   * :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` option -- automatically extends the Bluetooth buffers, which allows to speed up the FOTA transfer over Bluetooth, but also increases RAM usage.
```
```suggestion
   * The :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` Kconfig option -- automatically extends the Bluetooth buffers, which allows to speed up the FOTA transfer over Bluetooth, but also increases RAM usage.
```

peknis commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -34,17 +34,38 @@ You can also use FOTA updates to replace the application.
+   * SMP BT reassembly feature.
+   * :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP` option -- automatically extends the Bluetooth buffers, which allows to speed up the FOTA transfer over Bluetooth, but also increases RAM usage.
+
+If necessary, any of the implied options or defaulted values introduced by :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` can be modified.
```
```suggestion
If necessary, you can modify any of the implied options or defaulted values introduced by the :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option.
```

peknis commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -34,17 +34,38 @@ You can also use FOTA updates to replace the application.
+
+If necessary, any of the implied options or defaulted values introduced by :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` can be modified.
+
+You can either add the above Kconfig options to your application configuration files or inline them in a project build command.
```
```suggestion
You can either add these Kconfig options to the configuration files of your application or have them inline in a project build command.
```

peknis commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -34,17 +34,38 @@ You can also use FOTA updates to replace the application.
+If necessary, any of the implied options or defaulted values introduced by :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` can be modified.
+
+You can either add the above Kconfig options to your application configuration files or inline them in a project build command.
+See how it can be done for ref:`peripheral_lbs` sample:
```
```suggestion
Here is an example of how you can build for the :ref:`peripheral_lbs` sample:
```

peknis commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -34,17 +34,38 @@ You can also use FOTA updates to replace the application.
+    west build -b *build_target* -- -DCONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU=y
 
-  After completing these steps, your application advertises the SMP Service with ``UUID 8D53DC1D-1DB7-4CD3-868B-8A527460AA84``.
+Once you have completed these steps, when you connect to the device, you will see the SMP Service enabled with the ``UUID 8D53DC1D-1DB7-4CD3-868B-8A527460AA84``.
```
I don't see any steps.
Maybe something like:
When you connect to the device after the build has completed, the SMP Service is enabled with ....

mkapala-nordic replied:
I added something like `... and the firmware has been programmed to it ...` to your proposal as it won't work without it, please check current version

peknis commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -34,17 +34,38 @@ You can also use FOTA updates to replace the application.
 
-  After completing these steps, your application advertises the SMP Service with ``UUID 8D53DC1D-1DB7-4CD3-868B-8A527460AA84``.
+Once you have completed these steps, when you connect to the device, you will see the SMP Service enabled with the ``UUID 8D53DC1D-1DB7-4CD3-868B-8A527460AA84``.
+If you want to improve your application by adding SMP Service to advertising data, take a look at how it is done in :ref:`zephyr:smp_svr_sample`.
```
```suggestion
If you want to add SMP Service to advertising data, refer to the :ref:`zephyr:smp_svr_sample`.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -55,12 +51,8 @@ The build targets of interest for Thingy:53 in the |NCS| are listed in the follo
-   The |NCS| samples and applications that are compatible with the Nordic Thingy:53 follow the :ref:`thingy53_app_guide`.
-   The application guide defines a consistent partition map and bootloader configuration to allow serial and OTA firmware updates.
-
 The |NCS| by default uses :ref:`ug_multi_image` for Thingy:53.
```
```suggestion
The |NCS| uses :ref:`ug_multi_image` for Thingy:53 by default.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -55,12 +51,8 @@ The build targets of interest for Thingy:53 in the |NCS| are listed in the follo
-
 The |NCS| by default uses :ref:`ug_multi_image` for Thingy:53.
-Because of this, when you choose ``thingy53_nrf5340_cpuapp`` or ``thingy53_nrf5340_cpuapp_ns`` as the build target when building a sample or application that is compatible with Thingy:53, you will generate firmware for both the application core and network core:
+Because of this, when you choose ``thingy53_nrf5340_cpuapp`` or ``thingy53_nrf5340_cpuapp_ns`` as the build target when building a sample or application, you will generate firmware for both the application core and network core:
```
```suggestion
When you choose ``thingy53_nrf5340_cpuapp`` or ``thingy53_nrf5340_cpuapp_ns`` as the build target when building a sample or application, you will generate firmware for both the application core and network core:
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -127,7 +118,8 @@ To build and program the source code from the command line, complete the followi
 
-   The build target should be ``thingy53_nrf5340_cpuapp`` or ``thingy53_nrf5340_cpuapp_ns`` when building samples for the application core and ``thingy53_nrf5340_cpunet`` when building for the network core.
+   The build target should be ``thingy53_nrf5340_cpuapp`` or ``thingy53_nrf5340_cpuapp_ns`` when building samples for the application core.
+   The proper child image for ``thingy53_nrf5340_cpunet`` will be built automatically, check :ref:`thingy53_build_pgm_targets` for details.
```
```suggestion
   The proper child image for ``thingy53_nrf5340_cpunet`` will be built automatically.
   See :ref:`thingy53_build_pgm_targets` for details.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -156,57 +148,44 @@ See :ref:`thingy53_gs_updating_firmware` for details about updating firmware ima
 
-If you are using an external debug probe, such as the nRF5340 DK, or any J-Link device supporting ARM Cortex-M33, you do not have to use applications that follow the :ref:`thingy53_app_guide`.
-In that case, you can program the Thingy:53 in a similar way as nRF5340 DK.
+If you are using an external debug probe, such as the nRF5340 DK, or any J-Link device supporting ARM Cortex-M33, you can program the Thingy:53 in a similar way as nRF5340 DK.
```
```suggestion
If you are using an external debug probe, such as the nRF5340 DK, or any J-Link device supporting ARM Cortex-M33, you can program the Thingy:53 the same way as nRF5340 DK.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -156,57 +148,44 @@ See :ref:`thingy53_gs_updating_firmware` for details about updating firmware ima
 ----------------------------------------
 
-The Thingy:53-compatible samples and applications include the MCUboot bootloader that you can use to update firmware.
+Samples and applications built for Thingy:53 include the MCUboot bootloader that you can use to update firmware out of the box.
```
```suggestion
Samples and applications built for Thingy:53 include the MCUboot bootloader that you can use to update the firmware out of the box.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -156,57 +148,44 @@ See :ref:`thingy53_gs_updating_firmware` for details about updating firmware ima
   In this scenario Thingy is connected directly to your PC through USB.
 * Update the firmware over-the-air (OTA) using Bluetooth LE and the nRF Programmer mobile application for Android or iOS.
-  To use this method, the application that is currently programmed on Thingy:53 must support it.
+  To use this method, the application that is currently programmed on Thingy:53 must support it, refer to :ref:`thingy53_app_fota_smp` section.
```
```suggestion
  To use this method, the application that is currently programmed on Thingy:53 must support it.
  For details, refer to :ref:`thingy53_app_fota_smp` section.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -156,57 +148,44 @@ See :ref:`thingy53_gs_updating_firmware` for details about updating firmware ima
-
-   All applications compatible with the Nordic Thingy:53 in the |NCS| also support the additional features.
-   Refer to the source code of the samples and applications for examples of implementation.
+Because of that, Thingy:53 board by default enables MCUboot bootloader with serial recovery support and predefined static Partition Manager memory map.
+It also gives you an option to manually enable FOTA updates.
+Please refer to the following sections for details of what is configured by default and what you can configure by yourself.
```
```suggestion
Because of that, the Thingy:53 board enables MCUboot bootloader with serial recovery support and predefined static Partition Manager memory map by default.
You can also enable FOTA updates manually.
See the following sections for details of what is configured by default and what you can configure by yourself.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -156,57 +148,44 @@ See :ref:`thingy53_gs_updating_firmware` for details about updating firmware ima
 ===============================
 
-The samples compatible with the Nordic Thingy:53 use :ref:`partition_manager` to define memory partitions.
+The samples and applications for Nordic Thingy:53 use :ref:`partition_manager` by default to define memory partitions.
```
```suggestion
The samples and applications for Nordic Thingy:53 use the :ref:`partition_manager` by default to define memory partitions.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -156,57 +148,44 @@ See :ref:`thingy53_gs_updating_firmware` for details about updating firmware ima
+The samples and applications for Nordic Thingy:53 use :ref:`partition_manager` by default to define memory partitions.
 The memory layout must stay consistent, so that MCUboot can perform proper image updates and clean up the settings storage partition.
-To ensure that partition layout does not change between builds, the sample must use a static partition layout that is consistent between all compatible samples in the |NCS|.
+To ensure that partition layout does not change between builds, the sample must use a static partition layout that is consistent between all samples in the |NCS|.
```
```suggestion
To ensure that the partition layout does not change between builds, the sample must use a static partition layout that is consistent between all samples in the |NCS|.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -220,48 +199,43 @@ The overlay is applied automatically.
 ==================
 
-Each sample must enable the MCUboot bootloader and use the same configuration of the bootloader as all of the Nordic Thingy:53-compatible samples in the |NCS|.
+MCUboot bootloader is enabled by default for Thingy:53 in board's :file:`Kconfig.defconfig` file.
```
```suggestion
MCUboot bootloader is enabled by default for Thingy:53 in the :file:`Kconfig.defconfig` file of the board.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -220,48 +199,43 @@ The overlay is applied automatically.
 
-An application compatible with the Nordic Thingy:53 is expected to enable USB CDC ACM as backend for logging.
-The logs are provided using USB CDC ACM to allow accessing them without additional hardware.
+The logs on Nordic Thingy:53 board are, by default, provided using USB CDC ACM to allow accessing them without additional hardware.
```
```suggestion
The logs on the Thingy:53 board are provided by default using USB CDC ACM to allow access to them without additional hardware.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -220,48 +199,43 @@ The overlay is applied automatically.
 If a sample supports additional USB classes or more than one instance of USB CDC ACM, it must use a dedicated product name, vendor ID, and product ID.
 
-To enable the USB device stack and the CDC ACM class:
+The :kconfig:option:`CONFIG_BOARD_SERIAL_BACKEND_CDC_ACM` Kconfig option (defined in :file:`zephyr/boards/arm/thingy53_nrf5340/Kconfig.defconfig`) automatically sets default values of USB product name, vendor ID and product ID of Thingy:53.
+It also enables USB device stack and initializes USB device at boot.
+USB device remote wakeup feature is disabled by default as it requires extra action from the application side.
+A single USB CDC ACM instance is automatically included in the default board's DTS configuration (:file:`zephyr/boards/arm/thingy53_nrf5340/thingy53_nrf5340_common.dts`).
+The USB CDC instance is used to forward application logs.
```
```suggestion
The :kconfig:option:`CONFIG_BOARD_SERIAL_BACKEND_CDC_ACM` Kconfig option (defined in the :file:`zephyr/boards/arm/thingy53_nrf5340/Kconfig.defconfig` file) automatically sets the default values of USB product name, vendor ID and product ID of Thingy:53.
It also enables the USB device stack and initializes the USB device at boot.
The remote wakeup feature of a USB device is disabled by default as it requires extra action from the application side.
A single USB CDC ACM instance is automatically included in the default board's DTS configuration file (:file:`zephyr/boards/arm/thingy53_nrf5340/thingy53_nrf5340_common.dts`).
The USB CDC instance is used to forward application logs.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -220,48 +199,43 @@ The overlay is applied automatically.
-* Use the options :kconfig:option:`CONFIG_USB_DEVICE_STACK` and :kconfig:option:`CONFIG_USB_CDC_ACM`.
-* Set the default log level with the :kconfig:option:`CONFIG_USB_CDC_ACM_LOG_LEVEL_ERR` and :kconfig:option:`CONFIG_USB_DEVICE_LOG_LEVEL_ERR` Kconfig options.
-* Enable the UART log backend with the :kconfig:option:`CONFIG_LOG_BACKEND_UART` Kconfig option and disable SEGGER RTT with both the :kconfig:option:`CONFIG_LOG_BACKEND_RTT` and :kconfig:option:`CONFIG_USE_SEGGER_RTT` Kconfig options.
+If you do not want to use the USB CDC ACM as a backend for logging out of the box, you can disable it:
```
```suggestion
If you do not want to use the USB CDC ACM as a backend for logging out of the box, you can disable it as follows:
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -220,48 +199,43 @@ The overlay is applied automatically.
 
-See :file:`thingy53_nrf5340_cpuapp.conf` in a compatible sample for how this is done.
+* Disable the :kconfig:option:`CONFIG_BOARD_SERIAL_BACKEND_CDC_ACM` Kconfig option.
+* If USB CDC ACM is not being used for anything else, it can also be disabled in the application DTS overlay:
```
```suggestion
* If USB CDC ACM is not used for anything else, you can disable it in the application's DTS overlay file:
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -220,48 +199,43 @@ The overlay is applied automatically.
+* If USB CDC ACM is not being used for anything else, it can also be disabled in the application DTS overlay:
 
-In addition, you have to call the :c:func:`usb_enable` function from your application to initialize the USB stack.
+.. code-block:: none
```
```suggestion
  .. code-block:: none
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -220,48 +199,43 @@ The overlay is applied automatically.
+.. code-block:: none
 
-See the :file:`thingy53.c` source file in a compatible sample for how this is done.
+   &cdc_acm_uart {
+		status = "disabled";
+	};
```
```suggestion
     &cdc_acm_uart {
		  status = "disabled";
	  };
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -295,49 +269,29 @@ FOTA updates on the Nordic Thingy:53 are supported using MCUmanager's Simple Man
-The application must also call :c:func:`smp_bt_register` to register the SMP Bluetooth Service.
-
-See the :file:`thingy53.c` source file in a compatible sample for how this is done.
+To enable FOTA feature, use :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option.
+For details about implementation, please take a look at :ref:`ug_nrf52_developing_ble_fota` guide.
```
```suggestion
To enable the FOTA feature, use the :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option.
For details about the implementation, see the :ref:`ug_nrf52_developing` guide.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -295,49 +269,29 @@ FOTA updates on the Nordic Thingy:53 are supported using MCUmanager's Simple Man
-See the :file:`thingy53.c` source file in a compatible sample for how this is done.
+To enable FOTA feature, use :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option.
+For details about implementation, please take a look at :ref:`ug_nrf52_developing_ble_fota` guide.
+Note that MCUboot is already enabled by default on Thingy:53 so that there is no need to do it manually in application or sample configuration.
+Also, Thingy:53 supports network core upgrade out of the box.
```
```suggestion
MCUboot is already enabled by default on Thingy:53, so you do not need to do it manually in the application or sample configuration.
Thingy:53 supports network core upgrade out of the box.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -295,49 +269,29 @@ FOTA updates on the Nordic Thingy:53 are supported using MCUmanager's Simple Man
-This means that a sample compatible with the Nordic Thingy:53 must enable the external flash using the QSPI driver.
-Enable the option :kconfig:option:`CONFIG_NORDIC_QSPI_NOR`, and set :kconfig:option:`CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE` to ``4096`` and :kconfig:option:`CONFIG_NORDIC_QSPI_NOR_STACK_WRITE_BUFFER_SIZE` to ``16``.
-See :file:`thingy53_nrf5340_cpuapp.conf` in a compatible sample for how this is done.
+Thingy:53 board automatically configures external flash storage and QSPI driver when FOTA updates are implied with :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option.
```
```suggestion
The Thingy:53 board automatically configures external flash storage and QSPI driver when FOTA updates are implied with the :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option.
```

peknis commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -295,49 +269,29 @@ FOTA updates on the Nordic Thingy:53 are supported using MCUmanager's Simple Man
+===============================================================
 
-The following samples and applications in the |NCS| are already compatible with Thingy:53:
+The following samples and applications in the |NCS| already enable by default FOTA for Thingy:53:
```
```suggestion
The following samples and applications in the |NCS| enable FOTA for Thingy:53 by default:
```

mkapala-nordic commented on doc/nrf/releases/release-notes-changelog.rst:
```diff
@@ -52,6 +52,27 @@ Build system
+      See details in :ref:`thingy53_app_fota_smp` section of Thingy:53 application guide.
+    * Enabled by default MCUboot bootloader in Thingy:53 board configuration.
+      See details in :ref:`thingy53_app_mcuboot_bootloader` section of Thingy:53 application guide.
+    * Cleaned up configuration files of Thingy:53 compatible samples and applications as a result of introducing the above simplifications.
```
I forgot to change `compatible` from here as there is no such term as `Thingy:53 compatible samples` anymore.
Please check if current version is acceptable

mkapala-nordic commented on doc/nrf/working_with_nrf/nrf52/developing.rst:
```diff
@@ -34,17 +34,38 @@ You can also use FOTA updates to replace the application.
 
+FOTA updates are supported using MCUmanager's Simple Management Protocol (SMP) over Bluetooth.
+The application acts as a GATT server and allows the connected Bluetooth Central to perform the firmware update.
+Sample must support Bluetooth peripheral role (:kconfig:option:`CONFIG_BT_PERIPHERAL`) in order to use this feature.
```
It refers to all samples so applied your change

mkapala-nordic commented:
@peknis I think that I addressed all your comments either in changes or in comments under conversations.

Please also take a look at this discussion, maybe you will have some idea how it could be resolved: https://github.com/nrfconnect/sdk-nrf/pull/10256#discussion_r1105847614
In a nutshell, there is an idea to share FOTA docs across nRF52, nRF53 AND also Thingy:53, but each of this platform have some minor differences when enabling/infos only for this platform.

nordicjm commented on doc/nrf/working_with_nrf/nrf53/thingy53.rst:
```diff
@@ -220,48 +201,43 @@ The overlay is applied automatically.
+  .. code-block:: none
 
-See the :file:`thingy53.c` source file in a compatible sample for how this is done.
+     &cdc_acm_uart {
+		  status = "disabled";
+	  };
```
Alignment of this is a bit wild, probably need something like:
```
    &cdc_acm_uart {
        status = "disabled";
    };
```

divipillai reviewed (commented):
Nits. Rest all looks good.

mkapala-nordic commented:
@divipillai, @nordicjm - addressed your comments.
